### PR TITLE
Reimplement getPageLayoutBox API for ad components

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: extensions/amp-auto-ads/0.1/placement.js [hunk 1]
```diff
@@ -158,9 +158,9 @@ export class Placement {
    * @return {!Promise<number>}
    */
   getEstimatedPosition() {
-    return getElementLayoutBox(this.anchorElement_).then((layoutBox) => {
-      return this.getEstimatedPositionFromAchorLayout_(layoutBox);
```
Comment (samouri) [inline extensions/amp-auto-ads/0.1/placement.js:162]: nit: achor --> anchor

**Comment by samouri** [comment]
This looks great!

File: src/utils/page-layout-box.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  }
+  const {offsetWidth, offsetHeight} = element;
+  return layoutRectLtwh(left, top, offsetWidth, offsetHeight);
+}
+
+/**
+ * @param {!Element} element
+ * @return {!Promise<!../layout-rect.LayoutRectDef>}
+ */
+export function measurePageLayoutBox(element) {
```
Comment (samouri) [inline src/utils/page-layout-box.js:41]: nit: why not name this more in line with the function directly above? e.g. `getPageLayoutBox` or `getPageLayoutBoxAsync`

Comment (dvoytenko) [reply src/utils/page-layout-box.js:41]: I was actually looking for new distinct naming. But would be happy to rename to `getPageLayoutBox(Async)` if you think that works better. I used "measure" to rhyme with `vsync.measure()`.

**Review by jridgewell — APPROVED** [review]

File: src/service/mutator-impl.js [hunk 3]
```diff
@@ -397,7 +397,7 @@ export class MutatorImpl {
     opt_callback
   ) {
     resource.resetPendingChangeSize();
-    const layoutBox = resource.getPageLayoutBox();
```
Comment (samouri) [inline src/service/mutator-impl.js:400]: How do we know that this line is safe?

File: src/service/resource.js [hunk 4]
```diff
@@ -540,6 +540,17 @@ export class Resource {
     this.element.updateLayoutBox(newBox, sizeChanges);
   }
 
+  /**
+   * Yields when the resource has bee measured.
```
Comment (samouri) [inline src/service/resource.js:544]: nit: bee --> been

**Review by samouri — APPROVED** [review]
Code LGTM. Would like a test

**Pull request merged**